scipy/optimize/_numdiff.py

```diff
@@ -344,34 +365,81 @@ def fun_wrapped(x):
     if np.any((x0 < lb) | (x0 > ub)):
         raise ValueError("`x0` violates bound constraints.")
 
-    h = _compute_absolute_step(rel_step, x0, method)
+    if as_linear_operator:
+        if rel_step is None:
+            rel_step = _compute_relative_step(method)
+
+        return _linear_operator_difference(fun_wrapped, x0,
```

Comment:
Nikolay, I have take some time to read the analysis on Numerical Recipes and It is not very easy to generalize it for our case. On the other hand, if you take as reference the deduction (that yield the same result) on p. 195-196 on Nocedal and Wright [book](http://www.bioinfo.org.cn/~wangchao/maa/Numerical_Optimization.pdf), it is much easier to generalize to our case.

Let ``f: R^n -> R^m``, using Taylor expansion:
```
f(x+h*p) = f(x) + h*J(x).dot(p) + h^2*O(||p||^2)
```
therefore:
```
J(x).dot(p)  = (f(x+h*p) - f(x))/h + s, for |s| <= L*h*||p||^2 
```
which is our equivalent of equation (8.4).

Using similar reasoning we could easily get something equivalent to eq. (8.5):
```
L*h*||p||^2 + 2*u*L_f/h
```
And therefore the optimal value of ``h``would be:
```
h = sqrt(2L_f/L*u/||p||^2) = sqrt(2L_f/L)* sqrt(u)/||p||
```
If we do as in the book and consider ``2L_f/L`` approximately equals to 1, we would get:
```
h = sqrt(u)/||p||
```